File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DistrictsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DistrictsRow } from "./DistrictsRow";

export interface DistrictsColumns {
    DistrictId: Column<DistrictsRow>;
    Name: Column<DistrictsRow>;
    CityName: Column<DistrictsRow>;
}

export class DistrictsColumns extends ColumnsBase<DistrictsRow> {
    static readonly columnsKey = 'Tanimlamalar.Districts';
    static readonly Fields = fieldsProxy<DistrictsColumns>();
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Districts/DistrictsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.Districts")]
[BasedOnRow(typeof(DistrictsRow), CheckNames = true)]
public class DistrictsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DistrictId { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string CityName { get; set; }
}